src/Azos/Data/Schema.cs

```diff
@@ -22,546 +22,504 @@ namespace Azos.Data
-
-
-      public override string ToString()
+    }
+
+    /// <summary>
+    /// Returns a field def that matches the desired backed name for the specified target or null
+    /// </summary>
+    /// <param name="targetName">Target or null, if null any target assumed</param>
+    /// <param name="backendName">The name of the backend</param>
+    /// <param name="backendNameComparison">The string comparison to use against the backend name, OrdinalIgnoreCase is dflt</param>
+    /// <returns>The desired field or null</returns>
+    public FieldDef TryFindFieldByTargetedBackendName(
+                                              string targetName,
+                                              string backendName,
+                                              StringComparison backendNameComparison = StringComparison.OrdinalIgnoreCase)
+    {
+      return m_FieldDefs.FirstOrDefault(fd => fd.GetBackendNameForTarget(targetName).Equals(backendName, backendNameComparison));
+    }
+
+    public IEnumerator<Schema.FieldDef> GetEnumerator() => FieldDefs.GetEnumerator();
+
+    System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() => FieldDefs.GetEnumerator();
+
+    /// <summary>
+    /// Returns a SchemaAttribute that matches the supplied targetName, or if one was not defined then
+    ///  returns SchemaAttribute which matches any target or null
+    /// </summary>
+    public SchemaAttribute GetSchemaAttrForTarget(string targetName)
+    {
+      if (targetName.IsNotNullOrWhiteSpace())
       {
-          return "Schema(Name: '{0}', Count: {1})".Args(Name, m_FieldDefs.Count);
+        var atr = m_SchemaAttrs.FirstOrDefault(a => targetName.EqualsIgnoreCase(a.TargetName));
+        if (atr != null) return atr;
       }
+      return m_SchemaAttrs.FirstOrDefault(a => TargetedAttribute.ANY_TARGET.EqualsIgnoreCase(a.TargetName));
+    }
 
+    public override string ToString()
+    {
+      return "Schema(Name: '{0}', Count: {1})".Args(Name, m_FieldDefs.Count);
+    }
 
-      /// <summary>
-      /// Performs logical equivalence testing of two schemas
-      /// </summary>
-      public bool IsEquivalentTo(Schema other, bool compareNames = true)
-      {
-          if (other==null) return false;
-          if (object.ReferenceEquals(this, other)) return true;
+    /// <summary>
+    /// Performs logical equivalence testing of two schemas
+    /// </summary>
+    public bool IsEquivalentTo(Schema other, bool compareNames = true)
+    {
+      if (other == null) return false;
+      if (object.ReferenceEquals(this, other)) return true;
 
-          if (compareNames)
-            if (!Name.EqualsIgnoreCase(other.Name)) return false;
+      if (compareNames)
+        if (!Name.EqualsIgnoreCase(other.Name)) return false;
 
-          if (this.m_SchemaAttrs.Count != other.m_SchemaAttrs.Count ||
-              this.m_FieldDefs.Count != other.m_FieldDefs.Count) return false;
+      if (m_SchemaAttrs.Count != other.m_SchemaAttrs.Count ||
```

Comment:
I would keep 'this.' in any method that juxtaposes fields side-by side for clarity